rasterio/_features.pyx

```diff
@@ -284,8 +284,8 @@ def _rasterize(shapes, image, transform, all_touched, merge_alg_add):
     try:
         if all_touched:
             options = CSLSetNameValue(options, "ALL_TOUCHED", "TRUE")
-        if merge_alg_add:
-            options = CSLSetNameValue(options, "MERGE_ALG", "ADD")
+        options = CSLSetNameValue(
+            options, "MERGE_ALG", MergeAlg[merge_alg].value)
```

Comment:
@fredericbonifas GDAL functions need UTF-8 encoded strings. for Python 3, we'll need something like this:

```
merge_algorithm = MergeAlg[merge_alg].value.encode('utf-8')
options = CSLSetNameValue(options, "MERGE_ALG", <const char *>merge_algorithm)
```